### ISSUE TITLE
The package does not support the ISO 690:2021 standard

### ISSUE THREAD

**Issue by Witiko** [michal-h21/biblatex-iso690#127]
Apparently, the package `biblatex-iso690` does not yet support the ISO 690:2021 standard. @man-ka and I are planning to borrow ČSN ISO 690:2022 from a library to see the extent of changes compared to ČSN ISO 690:2011.

Should we open tickets with descriptions of the necessary changes to the package?

**Comment by michal-h21** [comment]
There already is [a pull request](https://github.com/michal-h21/biblatex-iso690/pull/124) with work in progress support for the new ISO 690 release. I haven't tested it yet, but I think it should be the start.

**Comment by Witiko** [comment]
Good to know. We will borrow the text of the new norm regardless, so we can test the code from the pull request #124 and see how well it complies.

**Comment by wg030** [comment]
> Good to know. We will borrow the text of the new norm regardless, so we can test the code from the pull request #124 and see how well it complies.

I have already improved some things on the initial PR #124, but I haven't uploaded them yet because I need to sort the new things a little and because I don't want to flood this repository with too many PR either .

Hence, until @michal-h21 has found some time to test the new style, I have created a temporary repository where I am going to commit my latest changes to (https://github.com/wg030/biblatex-iso690_2021).
This way you can test your bibliography against the latest updates. For example, I have already improved the patent entry type as well as the inbook entry type signifcantlly already. I will upload a beta version v0.2 soon to the temporary repository, so keep an eye on it.
Feedback on your side would be cool!

**Comment by wg030** [comment]
I have just uploaded the beta version v0.2 to the temporary repository! 🥳

**Comment by Witiko** [comment]
@wg030 Thanks for letting us know. Do you have some preferred timeline for receiving input on the beta version? :slightly_smiling_face:

**Comment by wg030** [comment]
> @wg030 Thanks for letting us know. Do you have some preferred timeline for receiving input on the beta version? :slightly_smiling_face: 

Well, since I work on that style rather along the way than regularly, there is no timeline. So feel free to simply post any feedback whenever you like. I will then consider it when I have time and energy to do so.

**Comment by michal-h21** [comment]
I've just installed your repo, and it is impressive! I don't see any problems at first glance, other than with the norm itself. I am unsure about using small caps for author names, I cannot find anything about it in the norm. They use uppercase in examples. 

I see that there is a lot of new language  strings, so we will need to translate them to Czech and other languages.

**Comment by wg030** [comment]
> I've just installed your repo, and it is impressive! I don't see any problems at first glance, other than with the norm itself. I am unsure about using small caps for author names, I cannot find anything about it in the norm. They use uppercase in examples. 
> 

As far as I remember the norm does not really tell how to emphasize names or titles. I personally think small caps names are a little less aggressive than uppcase names and that's why I prefer them.
Nevertheless, you can already switch to uppercase for your testing if that's your personal preference. Just go to line 280 in `iso690_2021.bbx`and remove the comment there and comment out line 281 instead. I actually wanted to make this a new package option, but it seems as if I have forgotten it again. No problem though, I will do in the next beta version.



> I see that there is a lot of new language  strings, so we will need to translate them to Czech and other languages.

Yeah, I created a lot of new language strings that were given in the norm implicitly. Concerning the translation I hope you noticed that I created an empty template for new languages (`newlang-iso690_2021.lbx`). I would leave the translation of other languages to persons that can speak the corresponding language perfectly, i.e. ideally native speakers. That's why only a German translation exists right now.

**Comment by vactomas** [comment]
Hi,
I wanted to ask, how is the support for the new standard going? Any new progress?

**Comment by michal-h21** [comment]
The progress happens in a separate repo: https://github.com/wg030/biblatex-iso690_2021 

There is a question as to whether the code should be imported here, or if a new CTAN package should be created, considering that the new package introduces its own options and does not share code with the original package.

@wg030 what to you think?

**Comment by wg030** [comment]
> Hi, I wanted to ask, how is the support for the new standard going? Any new progress?

As @michal-h21 said the development takes place in the repo he mentions. Right now the results of the new style seem to be quite good for let's say an "average bibliography". I occasionally improve things when I have some time such that even the more specific examples of the official document work out as expected.
However I would rather like to improve on let's say "real world" examples, i.e. entries of biliiographies that someone really wants to cite. Hence feedback of what works and what does not work yet is highly appreciated. Just open a new issue in the repo.

---



> There is a question as to whether the code should be imported here, or if a new CTAN package should be created, considering that the new package introduces its own options and does not share code with the original package.
> 
> @wg030 what to you think?

At some point in the fututure when enough beta testing has been done and the state of the new style can be considered as ready to go I would like to try to integrate the code here im some way. Though I haven't thought about how this good be done in an effective way I always thing that anything is possible.

**Comment by vactomas** [comment]
Sure thing, I'd be open to helping test it out. Right now, I use a different approach for my thesis, but it should be relatively straightforward to make the change.

My only limitation would be installation as I'd have to find a way to install it onto my selfhosted Overleaf instance and relatively limited time available. Not sure how difficult would it be to get it running and making sure it works for Czech.

**Comment by michal-h21** [comment]
@vactomas it should suffice to just copy files from the Github repository to your Overleaf project, if you want to test it just on one document.

**Comment by michal-h21** [comment]
@wg030 sounds good :)

**Comment by vactomas** [comment]
Just to make sure I understand it correctly. I copy over the iso690_2021.bbx, .cbx and .dbx into my Overleaf project and then translate [this .ltx file](https://github.com/wg030/biblatex-iso690_2021/blob/main/newlang-iso690_2021.lbx) to Czech?

**Comment by michal-h21** [comment]
@vactomas yes. It would be great if you could create `czech-iso690_2021.lbx` and send it to us, as we will need the Czech translation.

**Comment by vactomas** [comment]
Sure thing, I already forked the repo. I'll try my best to translate it, but I am not sure if I'll get it correct all the way.

**Comment by Zweihorn** [comment]
I highly appreciate this project.

Just for the benefit of clarity and reference:
  * [ISO 690:2021 - Information and documentation - Guidelines for bibliographic references and citations to information resources](https://www.iso.org/standard/72642.html?browse=ics)
  * [ DIN ISO 690:2013-10 - Information und Dokumentation - Richtlinien für Titelangaben und Zitierung von Informationsressourcen (ISO 690:2010) ](https://www.dinmedia.de/de/norm/din-iso-690/190805484)

Unfortunately, there seems to be some lag in Germany to produce a harmonised translation to the current ISO Standard.
